### PR TITLE
KEP-2771: Supports https for httpGet of Lifecycle

### PR DESCRIPTION
**Pull request by wzshiming**
Supports https for httpGet of Lifecycle

KEP #2771
PR kubernetes/kubernetes#102474

### CODE REVIEW

File: keps/sig-node/2771-pod-lifecycle-https/README.md [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+### Goals
+
+- Supports https for httpGet of Lifecycle
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
```
Comment (aojea) [inline keps/sig-node/2771-pod-lifecycle-https/README.md:89]: what about something like
```
as a Kubernetes user I want to be able to use https for my pod lifecycle hooks
```

Comment (wzshiming) [reply keps/sig-node/2771-pod-lifecycle-https/README.md:89]: Thanks, updated

File: keps/sig-node/2771-pod-lifecycle-https/README.md [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+How will security be reviewed, and by whom?
+How will UX be reviewed, and by whom?
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+Supports HTTPS scheme for httpGet of Lifecycle, and skip TLS verification.
```
Comment (khenidak) [inline keps/sig-node/2771-pod-lifecycle-https/README.md:109]: I am not sure we should support skip `tls verification`. At least it must not be the default behavior. Must be an opt in. 

If we assume that the reason of supporting TLS is security of the lifecycle hook, then it does not make sense to skip TLS verification.

Comment (aojea) [reply keps/sig-node/2771-pod-lifecycle-https/README.md:109]: See discussion here https://github.com/kubernetes/kubernetes/pull/102474#issuecomment-854704317

The other dimension is if we want to make http lifecycle hooks equal to probes, that does skip TLS verification ... this is a tough one

**Pull request closed without merging**